src/libraries/System.Linq/src/System/Linq/Reverse.cs

```diff
@@ -23,6 +23,21 @@ public static IEnumerable<TSource> Reverse<TSource>(this IEnumerable<TSource> so
             return new ReverseIterator<TSource>(source);
         }
 
+        public static IEnumerable<TSource> Reverse<TSource>(this TSource[] source)
+        {
+            if (source is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.source);
+            }
+
+            if (source.Length == 0)
```

Comment:
Nit: I would probably just have this method call the existing `Reverse` directly, although I see why you might have wanted to avoid a redundant type check.